Update Deployments as code docs (#1792)
astronomer/docs: astro/manage-deployments-as-code.md

---

After you create an Astro Deployment, you can create a file with the Astro CLI that contains its unique configurations represented as code. That includes worker queues, environment variables, and Astro Runtime version. You can use Deployment files to manage Deployments programmatically on Astro.

Specifically, you can:

- Create a template file in a central GitHub repository and use it as a source of truth for new Deployments that fit a particular use case. For example, you can standardize your team's development environments by creating a template file with configurations for that type of Deployment.
- Store a Deployment file that represents the configurations of an existing Deployment. You can make changes to this file to update a Deployment faster and more easily than doing so with the Cloud UI or individual Astro CLI commands.
- Use a template file from an existing Deployment to create another Deployment with the same configurations. This is an alternative to creating a new Deployment in the Cloud UI and manually copying configurations.

> Astro supports updating a Deployment programmatically with [Deployment API keys](api-keys.md), but you cannot currently automate creating a Deployment as part of a CI/CD workflow. Support for both of these features is coming soon.

Reviewer: ```suggestion
Astro supports updating a Deployment programmatically with [Deployment API keys](api-keys.md), but you can't  automate creating a Deployment as part of a CI/CD workflow.
```
We should try and avoid future looking statements. They can be a maintenance nightmare and could potentially cause embarrassment for the company.

Author: I agree wholeheartedly!

### Template file reference

> When you inspect a Deployment, aa template file is created with the following sections:

Reviewer: ```suggestion
When you inspect a Deployment, a template file is created with the following sections:
```

    alert_emails:
        - paola@cosmicenergy.io
        - viraj@cosmicenergy.io
# This section is only populated when inspecting a Deployment without creating a template. 
# Do not configure metadata when you create a new Deployment from a template. 
    metadata:
        deployment_id: 
        workspace_id: 
        cluster_id: 
        release_name: 
        airflow_version: 
        status: 
        created_at: 
        updated_at: 
        deployment_url: 
        webserver_url: 
```

> Read the following sections to learn about each available key and value in the template file.

Reviewer: ```suggestion

```
This should be obvious.  If not, link to the related sections, or add a _see_ statement after each entry.

   
   You can also go to the Workspace page in the Cloud UI.

> 4. Optional. Reconfigure any Airflow connections or variables from the Deployment which you copied into the template file. Airflow connections and variables cannot be configured using template files. See [Manage connections in Airflow](https://docs.astronomer.io/learn/connections).

Reviewer: ```suggestion
4. Optional. Reconfigure any Airflow connections or variables from the Deployment that you copied into the template file. Airflow connections and variables cannot be configured using template files. See [Manage connections in Airflow](https://docs.astronomer.io/learn/connections).
```
